pkg/cmd/admin/groups/sync/rfc2307/ldapinterface.go

```diff
@@ -57,6 +59,9 @@ type LDAPInterface struct {
 	// cachedUsers holds the result of user queries for later reference, indexed on user UID
 	// e.g. this will map an LDAP user UID to the LDAP entry returned from the query made using it
 	cachedUsers map[string]*ldap.Entry
+
+	// Err is used to log errors found interfacing with LDAP
+	Err io.Writer
```

Comment:
> What sort of logic do we want to have on which out of bounds errors are fatal? We could potentially also export the out of bounds error type from ldaputil and not have to pass around interface{} as the information you're interested in is mostly embedded in the error itself.

Keep this interface contained to this package for the time-being.  This isn't a particular common thing to have to do.

Context is required in the general case, so it should be part of the interface.